src/support/tx.js

```diff
@@ -400,44 +390,64 @@ tx.createPlatformCoinTransferTx = async function({
   txOptions = {}
 }) {
   // Merging txOptions
-  const updatedTxOptions = _.merge({}, { gasPrice: TX_GAS_PRICE, chainId: TX_CHAIN_ID }, txOptions);
-
-  // Generating nonce if required
-  let noncePromise = updatedTxOptions.nonce ? Promise.resolve(updatedTxOptions.nonce) : undefined;
-  if (!noncePromise) {
-    noncePromise = assignedNonce ? nonceManager.getNonceForAccount(fromAddress) : Promise.resolve();
+  const updatedTxOptions = _.defaults({}, txOptions, { gasPrice: TX_GAS_PRICE, chainId: TX_CHAIN_ID });
+
+  // determining tx nonce promise
+  let noncePromise = Promise.resolve();
+  let nonceReleasePromise = Promise.resolve();
+  if (updatedTxOptions.nonce) {
+    // Use nonce provided by client.
+    noncePromise = Promise.resolve(updatedTxOptions.nonce);
+  } else if (assignedNonce) {
+    // Let nonce manager to decide nonce value.
+    noncePromise = nonceManager.getNonceForAccount(fromAddress);
+    nonceReleasePromise = nonce => nonceManager.releaseAccountNonce(fromAddress, nonce);
   }
 
-  const createTransaction = nonce =>
-    withOptionalNonce(nonce, {
-      from: fromAddress,
-      to: toAddress,
-      value,
-      gas: '0x5208', // 21000
-      gasPrice: `0x${updatedTxOptions.gasPrice.toString(16)}`,
-      chainId: `0x${updatedTxOptions.chainId.toString(16)}`
-    });
+  const createTransaction = async nonce => {
+    try {
+      return withOptionalNonce(nonce, {
+        from: fromAddress,
+        to: toAddress,
+        value,
+        gas: '0x5208', // 21000
+        gasPrice: `0x${updatedTxOptions.gasPrice.toString(16)}`,
+        chainId: `0x${updatedTxOptions.chainId.toString(16)}`
+      });
+    } catch (error) {
+      logger.error(`Error during creating tx: ${error.message}`, error, {
+        fromAddress,
+        toAddress,
+        value,
+        updatedTxOptions
+      });
 
-  return noncePromise.then(createTransaction).catch(error => {
+      await nonceReleasePromise(nonce);
+      throw mapError(error);
+    }
+  };
+
+  try {
+    const nonce = await noncePromise;
+    return createTransaction(nonce);
+  } catch (error) {
     throw mapError(error);
-  });
+  }
 };
 
 /**
  * Returns the transaction receipt by transaction hash.
  * @param {string} txHash - A hash of the transaction to get receipt for.
  * @returns {Promise<TransactionReceipt|CvcError|Error>} A promise of the transaction receipt or error.
  */
-tx.getTransactionReceipt = function(txHash) {
-  return new Promise((resolve, reject) => {
-    tx.web3.eth.getTransactionReceipt(txHash, (error, result) => {
-      if (error) {
-        logger.error(`Error retrieving transaction receipt for ${txHash}.`, error);
-        return reject(mapError(error));
-      }
-      return resolve(result);
-    });
-  });
+tx.getTransactionReceipt = async function(txHash) {
+  const getTransactionReceiptPromise = util.promisify(cb => tx.web3.eth.getTransactionReceipt(txHash, cb));
+  try {
+    return await getTransactionReceiptPromise();
```

Comment:
`return await` is unnecessary
```suggestion
    return getTransactionReceiptPromise();
```
is the same